python/cudf/cudf/tests/test_struct.py

```diff
@@ -205,6 +205,16 @@ def test_dataframe_to_struct():
     df["a"][0] = 5
     assert_eq(got, expect)
 
+    # check that a non-string (but convertible to string) named column can be
+    # converted to struct
+    with pytest.warns(UserWarning, match="will be casted"):
```

Comment:
Only wrap the `to_struct()` call that issues the warning with this `pytest.warns` context manager.